# CS249-Assignment2

## Task 1
the files exist in the corresponding directory.
### 1.1 dbg.py
To run the script use the following command:

`./dbg.py -i <input-file> -k <value-of-k> -o <contigs-output-file> -g <graph-output-file>`

Example output:
```
>conitig\_1 length=1040
TAGCATGTCACTACACCAACACAAGGTTACCAGCCGCAAATATTCGCCTGATCAAGGCCATAAGGCAAATGCGGGCTACGCGAAGTACCTCCCTCATAGTTAACTGCACGTCTAAAGACGTAACTGCACGTCTAAAGACGTAACTGCACGTCTAAAGACGAGATGTAGAAAACCGACGCCATGGTCCTTGAGACGTCGACGGGCTTGCCAGTTAGTACGGAATATTGGGGGCCCGTGGAACACACCGCAGCGTCGGACCATCGGTACAAGGACTTAGCGATCCACGTTTGTACCAGCAGCCAAAGCACGAGACACGTTGGTTCAGACTTAGCTGTATCGTCAGAGTGAACTCATATCAGGGGAAACTATGTCGTATATATAGAAGACGTTCTATGTAGTACTCATCTAGTGCAGTTTCTATCCGAAACGGCGGGATCCACAATGTACCGAAGATTTTCTCTGTCGACCGCTATAGAGTCCTACCGCGGGTAACCTCCCCGACATCGCTTCTCATCCAGCTGAACGTTGTACAAAAACGTCGGGTCCCTGGAAGCCCATACCCAACCCACCCGCACTTATCATAGTCGAAAAGCAGGGAGCCCGCACCCTTAAATCGTAAATCCAGAGGTGCACCGAAGAGTTCGCCGTGACTATGAAAAATTGGATTGGCCCAGCTCCCCCGAAAGACTGGTCGTCCAGTACTGCCAGCCCCGGCGCGTATGCGGTGATGAAGGGACCCCGAGAAAGGCTAGCGATAATCCGAAGTCGCGGGATGCTAATCGATTCGGCAGAGGGTGAGCGGCATTCTGGTTTAGACGCCTTATTGAGGTCACCTTTCCATTCCCCCAGTTGGTCGTCCGCGGCAATCAAACCTGGAATCGTAGGAAATTATGGTCAGTCTCGAGTGGTATGATGCAAACAGCCTAAACAAGACACTCGTCTCGTAACGAGTAGTCTGTGTCAAAATTATCGTGATGGCAATGGGGGGGACATTGAGAGGTTGGGAAAAACCAAACTCCTTCTGAAAGACCGTGGACCGA
```

### 1.2 olc.py
To run the script use the following command:

`./olc.py <input-file> -o <output-file> -n <min-overlap> [--ont]`

--ont used when the reads are ONT


Example output:
```
>contig_1
GACGCCCTCCAGTTGTGTAGAAATTGCACCGGAATAAATTATGTGAACCGTGGGCCTCTATCTCAAACGGCCATCGCAGA
AGCGCGCAGGCGATGTGTTTGAGAGGGACGTCCTCCGGCATCGCACTCGGCTTCGGTTTCGTGACTAAACTTTTGCCTTT
CGGCGAGTAGACGAGGAAGTCTTTATCTTCGCTCACATGCTGCCAGTGGACCTAAACGGTAAAACACGATCAGAAGGTCA
TGGCGGGATTATGCGACCACGTCGCGGACTGTAACGTTGCGACACTCTGACCCCGGTATTTTCCTCGAGGGGCCCAGCAG
CAGGTGTTTTCAATGGGTCTGTTAGTGCACAGCTCATACGTGTTTGTCATTGCCTAAGAAATGGGACGGAATGGCACATT
CTCACGGGAACGGGCTGGAAGCTCGAGGAAATCAACCTGGGCCGGTACTTACGATCAGATCGATTCGTGTAGCTGCTTAA
ATAATTATATCCGACCAATACATACCGTCACCGGCAAACGGTGCTGCCTCTAACTAGGCGCGCTTGGGCAGGGTGGGCAG
GACCCGCTTGGTCACGATCATTACACATCGTCACGAGAACCTAGACGGACGTGCTGGATGTAAATCCGTAATTTGCCTCT
CTCGATATGGATCTGGGCGCACGTGAGGGTCGTCCTGTCTTTGGTGATGGCGGACTAAAAAGCTGTCGGTCTTAATATAA
GGCGACTGTTTTCCGCAATTCTTCACCTCTAATGCCGGCTACACCGCTGACGGAAGACCTCGCGTCTCTCTCGATAAACC
AGGCTTCTACCCGCAGATCTCTCCGCATAATCTAGCCCATTGGCAGGGGGTGAGCCCAGTGCATCTCCAGATTAAGCCGT
TATCGTTATCAAGTAGACGCCCTCCAGTTGTGTAGAAATTGCACCGGAATAAATTATGTGAACCGTGGGCCTCTATCTCA
AACGGCCATCGCAGAAGCGCGCAGGCGATGTGTT
>contig_2
CCTAAGAAATGGGACGGAATGGCACATTCTCACGGGAACGGGCTGGAAGCCAGAGGAAATCAACCTGGGCCGGTACTTAC
GATCAGATCGATTCGTGTAG
>contig_3
CTAGCACAATGACTCTTGTGACAATATTCAGTGTATCTCCGAATTGCCCTCGCTTTTAACTGCGCGTCACATGCTGCCAG
TGGACCTAAACGGTAAAACACTAGCACAATGACTCTTGTGACAATATTCAGTGTATCTCCGAATTGCCCTCGCTTTTAAC
TGCGCGTCACATGCTGCCAGTGGACCTAAACGGTAAAACGGACTAGCACAATGACTCTTGTGACAATATTCAGTGTATCT
CCGAATTGCCCTCGCTTTTAACTGCGCGTCACATGCTGCCAGTGGACCTAAACGGTAAA
>contig_4
TAAGCGAGGAGGTATCGCCCTACCTGGACTAGCACAATGACTCTTGTGACAATATTCAGTGTATCTCCGAATTGCCCTCG
CTTTTAACTGCGCGTCACATGGTAAGCGAGGAGGTATCGCCCTACCTGGACTAGCACAATGACTCTTGTGACAATATTCA
GTGTATCTCCGAATTGCCCTCGCTTTTAACTGCGCGTCAC
>contig_5
ATTGGTAAGCGAGGAGGTATCGCCCTACCTGGACTAGCACAATGACTCTTGTGACAATATTCAGTGTATCTCCGAATTGC
CCTCGCTTTTAACTGCGCGT
>contig_6
TATTGGTAAGCGAGGAGGTATCGCCCTACCTGGACTAGCACAATGACTCTTGTGACAATATTCAGTGTATCTCCGAATTG
CCCTCGCTTTTAACTGCGCG
```

### 1.3
#### 1.3.1
use dbg\_graph\_only.py to draw the graph

`./dbg_graph_only.py -i <input-file> -k <value-of-k> -g <graph-output-file>`

Example of the first 5 lines of an output:
```
H	VN:Z:1.0
S	S0	CAGGGGGTGAGCCCAGTGCATCTCCAGATTAAGCCGTTA
S	S1	TATCGCCCTACCTGGACTAGCACAATGACTCTTGTGACA
S	S2	GAAGCTCGAGGAAATCAACCTGGGCCGGTACTTACGATC
S	S3	GAGGGACGTCCTCCGGCATCGCACTCGGCTTCGGTTTCG
```

#### 1.3.2
The QUAST script is included in the `scripts.txt` because it ran on ibex.
#### 1.3.3
We just use the commands in the previous parts to get the results.
#### 1.3.3
We just use the QUAST command to get the result.


## Task 2
all the scripts are in the script.txt file.

To produce the results, just go to the corresponding file and run the code.

Otherwise, everything is explained in the report.


### 2.1
To assemble the genome run the hifiasm script from the `script.txt` file.

### 2.2
To run QUAST, BUSCO, Merqury, and Inspector run the corresponding script from the `script.txt` file.
